src/tests/ftest/soak/soak.py

```diff
@@ -0,0 +1,338 @@
+                if result != "COMPLETED":
+                    self.fail("Soak job: {} didn't complete as expected: {}".
+                              format(job, result))
+
+        except (DaosApiError, IorUtils.IorFailed) as error:
+            self.fail("Soak Test 2 Failed/n {}".format(error))
+        finally:
+            try:
+                os.remove(script)
+            finally:
+                pass
+
+    def test_soak_3(self):
+        """
+        Test ID: DAOS-2192
+        Test Description: this time try a dmg command combined with IOR run
+        Use Cases:
+        :avocado: tags=soak3
+        """
+
+        script1 = None
+        script2 = None
+        try:
+            # retrieve IOR job parameters
+            s3_job1_name = self.params.get("name", '/run/job1/')
+            s3_job1_nodes = self.params.get("nodes", '/run/job1/')
+            s3_job1_processes = self.params.get("process_per_node",
+                                                '/run/job1/')
+            s3_job1_spec = self.params.get("jobspec", '/run/job1/')
+            cmd = self.get_ior_job(s3_job1_spec, s3_job1_processes,
+                                   s3_job1_name)
+
+            # write out a slurm job script
+            output = os.path.join(self.tmpdir, s3_job1_name + "_results.out")
+            script1 = slurm_utils.write_slurm_script(self.tmpdir, s3_job1_name,
+                                                    output,
+                                                    int(s3_job1_nodes), [cmd])
+            # queue it up to run and register a callback to retrieve results
+            job_id = slurm_utils.run_slurm_script(script1)
+            slurm_utils.register_for_job_results(job_id, job_done, maxwait=3600)
+
+            # now do the dmg job
+            dmgcmds = DmgUtils.get_dmg_script("dmg1", self.params,
+                                              self.basepath,
+                                              self.workdir)
+
+            s3_job2_name = self.params.get("name", '/run/job3/')
+            s3_job2_nodes = self.params.get("nodes", '/run/job3/')
+            output = os.path.join(self.tmpdir, s3_job2_name + "_results.out")
+            script2 = slurm_utils.write_slurm_script(self.tmpdir, s3_job2_name,
+                                                    output, s3_job2_nodes,
+                                                    dmgcmds)
+            job_id = slurm_utils.run_slurm_script(script2)
+            slurm_utils.register_for_job_results(job_id, job_done, maxwait=3600)
+
+            # wait for all the jobs to finish
+            global _soak_results
+            while len(_soak_results) < 2:
+                time.sleep(10)
+
+            for job, result in _soak_results.iteritems():
+                if result != "COMPLETED":
+                    self.fail("Soak job: {} didn't complete as expected: {}".
+                              format(job, result))
```

Comment:
This also looks like it could be [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)ed out.